feature/gribi/ate_tests/hierarchical_weight_resolution_test/README.md

```diff
@@ -86,28 +84,29 @@ WCMP width of 16 nexthops:
 
     *   IPv4Entry 192.0.2.222/32 in default VRF, pointing to NextHopGroup(NHG#3)
         in default VRF, with 16 NextHops(NH#100, NH#101, ..., NH#115), all with
-        weight: 1, in default VRF:
+        weight: 16 except NHG#100 is of weight 1, in default VRF:
 
         *   NH#100 with weight:1, pointing to 192.0.2.18
 
-        *   NH#101 with weight:1, pointing to 192.0.2.22
+        *   NH#101 with weight:16, pointing to 192.0.2.22
 
         *   ...
 
-        *   NH#115 with weight:1, pointing to 192.0.2.79
+        *   NH#115 with weight:16, pointing to 192.0.2.79
 
 *   Validate with traffic:
 
     *   NH10: (1/32) * (3/8) = 1.171% traffic received by ATE port-2 VLAN 1
 
     *   NH11: (1/32) * (5/8) = 1.953% traffic received by ATE port-2 VLAN 2
 
+    *   NH100: (31/32) * (1/241) = 0.402% traffic received by ATE port-2 VLAN 3
+
     *   for each VLAN ID in 3...18:
 
-        *   NH: (31/32) * (1/16) ~ 6.05% traffic received by ATE port-2 VLAN ID
+        *   NH: (31/32) * (16/241) = 6.432% traffic received by ATE port-2 VLAN ID
 
-    *   A deviation of 0.5% is allowed for each VLAN for now, since we only test
-        for 2 mins.
+    *   A deviation of 0.2% is allowed for each VLAN for now.
```

Comment:
Why is this updated to 0.2%? can we instead make this 1% to replicate the current test config?

Reply:
@xw-g, this is based on our conversation in b/270468786. Infact, imo there shouldnt be a deviation and we should just increase the tolerance to 1.5%. @xw-g, thoughts?

Reply:
0.2% came from Xiao (not sure about the math behind though). It's not a deviation  in the common sense we use here, more like just a deviation (tolerance as you called it) allowed to pass (as i read this)
Btw, why 1.5%, why not 15%? :)

Reply:
0.2% came from Xiao (not sure about the math behind though)

- Well, the point I am trying to make is, even though it was originally defined in the readme as 0.5%, the code is looking for 1%. [Link to the main] (https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/ate_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go#L63). So my comment was, why not update the Readme to be 1% instead and not 0.2%. If this came from Xiao, is this in a bug, can you please point me there?

I am quoting the original bug discussion here for clarity and context. So by reducing this to 0.2%, we will add flakiness to the test results.

====
Hello Xiao

Need some clarification on TE-3.3 test. Currently the test uses only [256 flows](https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/ate_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go#L361). This being a hashing test, we either need to define more flows in the order of ~1000 or, we need to relax the pass thresholds which are [currently defined as 1%](https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/ate_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go#L63).

Why: Even with WCMP, the DUT relies on hashing to load share between different egress interfaces. When there are less number of flows and a very tight deviation to meet (1% in this case) between the different egress interfaces, there are high possibilities for the test results to be flaky. Please also note: The test is using only a single source address and 256 destination addresses. These are native IP flows with no payload. Hence, IXIA uses the payload protocol for IP headers to be constant as protocol 61. So there is not much of an entropy available to get proper hashing. Not sure what is the business requirement here. Any idea?

In the meantime, proposal is to:
Update test to generate ~1024 flows by changing destination subnet to /22.
====

Reply:
* I'm ok with change the deviation to tolerance to avoid confusion.
* The 0.2% current is a arbitrary number. However, it does have some reasoning. (31/32) * (1/241) ~ 0.402% is the the lowest percentage we want to validate. So, we prefer the tolerance lowered that than. So, we start with 0.2% and we can revisit this number as we learn more from operation.
* @sachendras our b/270468786 discussion stands. I think the point is: if the DUT has challenges meeting the 0.2%, let's create a deviation flag or make the 0.2% value from an input arguments (this probably is better). If it's about that the test does not generate enough flows, then let's modify the code to generate more flows with enough hashing.  
* @sachendras @vosipchu for this test, I think we can start with generating more IP flows, say 254 x 254 = 64K (from more sources and more destinations). Later, as we have more clarification on hashing requirement, we can come back to add more parameters (e.g. IPinIP, etc). WDYT?

Reply:
"The 0.2% current is a arbitrary number. However, it does have some reasoning. (31/32) * (1/241) ~ 0.402% is the the lowest percentage we want to validate. So, we prefer the tolerance lowered that than. So, we start with 0.2% and we can revisit this number as we learn more from operation."

So based on the calculations and the weight definitions, we will be expecting 0.402% traffic on an interface at max sometimes. If so, looks like deviation_ucmp_traffic_tolerance w/ 1.5% tolerance of the total ingress traffic will pass the test even w/ 0% traffic on an interface, isnt it, or may be I am wrong here? :) Just a thought, making iterative progress towards the actual working number would be difficult. Anyhow, LGTM on the deviation.

- However, can we please update this PR for 254 x 254 = 64K flows and have the threshold set at 0.3% instead in this PR. It is still well within the requirement of ~ 0.402%. Additional PRs, just add toil to everyone :). @xw-g but I will leave it on u for 0.2% v/s 0.3%. But adding number of flows will definitely help in this PR.

Reply:
Left it unchanged: 
for basic hierarchical weight, tolerance is 0.5
and hierarchical weight in boundary scenarios is 0.2

Reply:
@manan-patel can you please add 254 x 254 flows?

Reply:
Added one flow in the terms of outer destination IP, but 64K (254x254) flows in the terms of inner source and destination IP.